### ISSUE TITLE
Error with `ezkl.mock()`

### ISSUE THREAD

**Issue by finlytics-hub** [zkonduit/ezkl#398]
### Describe the bug

Unable to run `ezkl.mock()` for a model with PReLU activation and BatchNorm1d. Error as below:

```
thread '<unnamed>' panicked at 'row=2097146, usable_rows=0..2097146, k=21', /root/.cargo/git/checkouts/halo2-049b997cf7195aea/b154081/halo2_proofs/src/dev.rs:441:13

PanicException: row=2097146, usable_rows=0..2097146, k=21
```

### Expected behaviors

Mock run to be successfull

### Steps to reproduce the bug

- Train a PyTorch model with PReLU activation and BatchNorm1d
- Compile model using `ezkl.compile_model()`
- Generate & calibrate settings and witness files
- Run `ezkl.mock()` using the attached files

### Device and Operating System

OS: Ubuntu
Version: 22.04
Processor: Intel Haswell 48 vCPUs
Memory: 100 GB

### Additional Information

- python 3.10.12
- ezkl v 1.7.19
- pytorch v 2.1.0.dev20230727+cu121
- onnx v 1.13.1
- [files attached](https://github.com/zkonduit/ezkl/files/12269808/mock.zip)

**Comment by finlytics-hub** [comment]
same issue with ReLU and BatchNorm1d model
[model_22.zip](https://github.com/zkonduit/ezkl/files/12273169/model_22.zip)

**Comment by alexander-camuto** [comment]
ty -- have found the issue and will address :)

**Comment by finlytics-hub** [comment]
One thing to add: the above errors were with public input and output and hashed model params. No errors were faced when all three were public.

**Comment by finlytics-hub** [comment]
getting a similar error for `ezkl.mock()` with a much simpler model but with a `batch_size` of 890 in `settings.json`. Worked fine with a `batch_size` of 800 though.

```
thread '<unnamed>' panicked at 'row=131066, usable_rows=0..131066, k=17', /root/.cargo/git/checkouts/halo2-049b997cf7195aea/b154081/halo2_proofs/src/dev.rs:496:9

PanicException: row=131066, usable_rows=0..131066, k=17
```
[model_1.zip](https://github.com/zkonduit/ezkl/files/12302470/model_1.zip)
